### PR TITLE
"Tipsy-ness" level selector

### LINKED ISSUE #13: Add "tispy-ness level" selector component
Being able to query for cocktails based on how tipsy you want to be is this project's core feature. The component responsible for selecting the "tipsy-ness level" should thus stand out in the overall simple UI.

I thought of implementing a control similar to those usually found in smart home thermostat applications (cf. the following mockup).

thermostat ui mockup

### PR DESCRIPTION
**Pull request by Quaffel**
Any feedback on the overall concept/design is highly appreciated. 

This approach might be a little over-engineered, but it's quite robust and extremely customizable. If you have any suggestions on how this may be solved in a _much_ simpler way, however, please don't hesitate to share them with me.

Regarding the color scheme: As @lenaalaugh is working on a new design, I didn't mind leaving the primary color as-is.

Resolves #13.

### CODE REVIEW

**Comment by Quaffel** [comment]
@Jonasdoubleyou Haven't tested it on mobile or other low-power devices yet. Would be great if you could re-deploy once the PR is merged. There are still a few things we could tweak in case we're running into performance problems...

**Review by Jonasdoubleyou — APPROVED** [review]
looks good

**Review by lenaalaugh — CHANGES_REQUESTED** [review]
All in all very good. However, please remove the unused import DegreeAngle in circleselector.tsx

**Comment by Jonasdoubleyou** [comment]
If you are at it, you could also just add it to the form @Quaffel

**Review by Jonasdoubleyou — APPROVED** [review]

**Review by lenaalaugh — APPROVED** [review]

**Comment by Jonasdoubleyou** [comment]
> @Jonasdoubleyou Haven't tested it on mobile or other low-power devices yet. Would be great if you could re-deploy once the PR is merged. There are still a few things we could tweak in case we're running into performance problems...

@Quaffel https://drunk.wilms.ninja/

File: frontend/src/tipsyselector/circleselector.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    centerToPoint.x / centerToPointLength * layoutOptions.radius + center.x,
+                    centerToPoint.y / centerToPointLength * layoutOptions.radius + center.y
+                )
+            };
+        }
+
+        // Rectangle containing all elements. Offsets are relative to the viewport.
+        const targetDimensionsInViewport: DOMRect = event.currentTarget.getBoundingClientRect();
+        const cursorPosition: Vector = new CartesianPoint(
+            event.clientX - targetDimensionsInViewport.left,
```
Comment (Jonasdoubleyou) [inline frontend/src/tipsyselector/circleselector.tsx:121]: @Quaffel Ah, missed this: the [TouchEvent](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent) does not have a clientX and clientY. Instead you need to access `.touches[0]`, see [here](https://developer.mozilla.org/en-US/docs/Web/API/Touch/clientX)

**Comment by Quaffel** [comment]
@Jonasdoubleyou Thanks for the heads-up and the deployment. I'll come up with a fix.

**Pull request merged**